frontend/src/services/web3-provider/ethereum-provider.ts

```diff
@@ -66,13 +67,17 @@ export abstract class EthereumProvider extends EventEmitter implements IEthereum
         chainName: name,
         rpcUrls: [rpcUrl],
         blockExplorerUrls: [explorerUrl],
-        nativeCurrency: {
-          name: 'Ethereum',
-          symbol: 'ETH',
-          decimals: 18,
-        },
+        nativeCurrency: nativeCurrency
+          ? toRaw(nativeCurrency)
```

Comment:
Lest move this `toRaw` function call [here](https://github.com/beamer-bridge/beamer/pull/1330/files#diff-8aa979d6df2fe2634cc8483af13978d4bb68643da290dd7c781a791d247290fbR34) instead.
I guess it would be nice to keep the `ethereum-provider.ts` module less dependent on vue's internals.
Composables anyways are supposed to be strongly dependent on the vue's specifics.

Reply:
Actually, i am just trying this out now, seems like it will not work if we move the `toRaw`  to the composable. So ignore this for now.